src/IssueViz/Editor/QuickActions/QuickFixes/QuickFixActionsSource.cs

```diff
@@ -0,0 +1,128 @@
+    {
+        private readonly ILightBulbBroker lightBulbBroker;
+        private readonly ITextView textView;
+        private readonly IThreadHandling threadHandling;
+        private readonly ITagAggregator<IIssueLocationTag> issueLocationsTagAggregator;
+
+        public QuickFixActionsSource(ILightBulbBroker lightBulbBroker,
+            IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService,
+            ITextView textView)
+            : this(lightBulbBroker, bufferTagAggregatorFactoryService, textView, new ThreadHandling())
+        {
+        }
+
+        internal QuickFixActionsSource(ILightBulbBroker lightBulbBroker,
+            IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService,
+            ITextView textView,
+            IThreadHandling threadHandling)
+        {
+            this.lightBulbBroker = lightBulbBroker;
+            this.textView = textView;
+            this.threadHandling = threadHandling;
+
+            issueLocationsTagAggregator = bufferTagAggregatorFactoryService.CreateTagAggregator<IIssueLocationTag>(textView.TextBuffer);
+            issueLocationsTagAggregator.TagsChanged += TagAggregator_TagsChanged;
+        }
+
+        public event EventHandler<EventArgs> SuggestedActionsChanged;
+
+        public IEnumerable<SuggestedActionSet> GetSuggestedActions(
+            ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range,
+            CancellationToken cancellationToken)
+        {
+            var allActions = new List<ISuggestedAction>();
+
+            if (IsOnIssueWithQuickFixes(range, out var issuesWithFixes))
+            {
+                foreach (var issueViz in issuesWithFixes)
+                {
+                    allActions.AddRange(
+                        (issueViz.Issue as IAnalysisIssue).Fixes
+                        .Select(fix => new QuickFixSuggestedAction("TODO fix message")));
+                }
+            }
+
+            return allActions.Any()
+                ? new[] { new SuggestedActionSet(allActions) }
+                : Enumerable.Empty<SuggestedActionSet>();
+        }
+
+        public async Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+        {
+            var hasActions = false;
+
+            await threadHandling.RunOnUIThread(() => hasActions = IsOnIssueWithQuickFixes(range, out _));
+
+            return hasActions;
+        }
+
+        private bool IsOnIssueWithQuickFixes(SnapshotSpan range, out IEnumerable<IAnalysisIssueVisualization> issuesWithFixes)
+        {
+            var tagSpans = issueLocationsTagAggregator.GetTags(range);
+
+            issuesWithFixes = tagSpans
+                .Select(x => x.Tag.Location)
+                .OfType<IAnalysisIssueVisualization>()
+                .Where(x => x.Issue is IAnalysisIssue analysisIssue && analysisIssue.Fixes.Any());
```

Comment:
We haven't established yet what the API would be, so I'm assuming that it will be some kind of a collection on the AnalysisIssue